TidepoolKit/Model/Datum/BasalDatum/TBasalDatum.swift

```diff
@@ -33,16 +33,16 @@ public class TBasalDatum: TDatum {
         guard self.deliveryType == expectedDeliveryType else {
             throw DecodingError.dataCorruptedError(forKey: .deliveryType, in: container, debugDescription: "Unexpected deliveryType '\(self.deliveryType)'")
         }
-        self.duration = try container.decodeIfPresent(Int.self, forKey: .duration)
-        self.expectedDuration = try container.decodeIfPresent(Int.self, forKey: .expectedDuration)
+        self.duration = try container.decodeIfPresent(Int.self, forKey: .duration).map { .milliseconds($0) }
```

Comment:
Q: why round to milliseconds?

Reply:
@rickpasetto The API specification states that the `duration` field (and `expectedDuration`) is a milliseconds-based integer. Therefore, we decode the integer and then convert it to a `TimeInterval` in milliseconds.

Reply:
@darinkrauss Gotcha. 👍